rskj-core/src/test/java/co/rsk/peg/ReleaseTransactionBuilderTest.java

```diff
@@ -904,6 +909,112 @@ void test_VerifyTXFeeIsSpentEquallyForBatchedPegouts_three_pegouts() {
+        Federation fed = FederationTestUtils.getFederation(memberPKs);
+        // list of peg-out requests with the given minimum peg-out value
+        List<ReleaseRequestQueue.Entry> entries = Arrays.asList(
+            createTestEntry(1000, minPegoutValue));
+        // this is a bit hacky, but this is to ensure the minPegoutValue chosen
+        // is not dust with the current wallet implementation. Either way this would be
+        // checked later on when building the transaction, but I think is good to have this explicitly
+        BtcTransaction tx = new BtcTransaction(networkParameters);
+        tx.addOutput(entries.get(0).getAmount(), entries.get(0).getDestination());
+        assertFalse(tx.getOutput(0).isDust());
+        // now we need to calculate the min value that an UTXO can use to be spent. This would be the least
+        // amount needed by the change ouput back to the federation address. This could also be the mininum 
+        // pegin value, but since a change output is also an UTXO that the current wallet implementation can use
+        // then we will find and use that value.
+        Coin minSpendableUTXOValue = Coin.SATOSHI;
+        tx.clearOutputs();
+        tx.addOutput(minSpendableUTXOValue, fed.getP2SHScript());
+        while (tx.getOutput(0).isDust()) {
+            minSpendableUTXOValue = minSpendableUTXOValue.add(Coin.SATOSHI);
+            tx.clearOutputs();
+            tx.addOutput(minSpendableUTXOValue, fed.getP2SHScript());
+        }
+        // list of utxos that contains N mimUnspendableUTXO to cover the minPegoutValue
+        List<UTXO> utxos = new ArrayList<>();
+        Coin[] neededUTXOsInCoin = minPegoutValue.divideAndRemainder(minSpendableUTXOValue.getValue());
+        long neededUTXOs = neededUTXOsInCoin[0].getValue();
+        if (!neededUTXOsInCoin[1].isZero()) {
+            neededUTXOs++;
+        }
+        for (int i = 0; i < neededUTXOs; i++) {
+          utxos.add(
+            new UTXO(mockUTXOHash("utxo " + i), 0, minSpendableUTXOValue, 0, false, fed.getP2SHScript()));
+        }
+        // the federation wallet, without flyover support
+        Wallet fedWallet = BridgeUtils.getFederationSpendWallet(
+            Context.getOrCreate(networkParameters),
+            fed,
+            utxos,
+            false,
+            mock(BridgeStorageProvider.class)
+        );
+        // build release transaction builder for current fed
+        when(activations.isActive(ConsensusRule.RSKIP201)).thenReturn(true);
+        ReleaseTransactionBuilder releaseTransactionBuilder = new ReleaseTransactionBuilder(
+            networkParameters,
+            fedWallet,
+            fed.getAddress(),
+            feePerKb,
+            activations
+        );
+
+        // build batch peg-out transaction
+        ReleaseTransactionBuilder.BuildResult result = releaseTransactionBuilder.buildBatchedPegouts(entries);
```

Comment:
With the current feePerKb of `24,000 SATS` and my initial proposed minPegoutValue of `28,500 SATS`, this batch peg-out transaction will not be built with a `Wallet.CouldNotAdjustDownwards` exception. Meaning the user output could not be adjusted downwards to pay the tx fees. This makes sense since this scenario I believe is the worst case scenario, since we have N P2SH inputs with the min possible value to be non-dust and only one user output that is going to take all the load. So I think if we can find two values that pass this scenario, then we should be good for any other scenario. Wdyt @marcos-iov ?

Reply:
Nevermind, this test scenario will never pass no matter what min peg-out value you choose. Maybe we can just split the min peg-out value by using 10 inputs. Like if min peg-out value is 20 then have 10 inputs of 2 each. Wdyt?

Reply:
Not sure I followed the logic here. There is only one pegout request for the min pegout value, right? And how many UTXOs available and for what value?

Reply:
Yes only one pegout request. And there will be as many UTXOs available to cover the request. Each UTXO will have a value of 2,700 SATS. The problem with this test is that is impossible to pass since it is too much fees that has to be taken from the user. I have written a similar one but that takes the pegout request value and divides it by 10 equal UTXOs.